server/src/main/java/org/opensearch/index/shard/IndexShard.java

```diff
@@ -1359,6 +1360,21 @@ public GatedCloseable<IndexCommit> acquireLastIndexCommit(boolean flushFirst) th
         }
     }
 
+    public void finalizeReplication(SegmentInfos infos, MetadataSnapshot expectedMetadata, long seqNo) throws IOException {
+        if (getEngine() instanceof NRTReplicationEngine) {
```

Comment:
I think we should throw if the wrong engine is configured.

```suggestion
        assert getEngine() instanceof NRTReplicationEngine;
        ((NRTReplicationEngine) getEngine()).updateSegments(infos, seqNo);
```